community/detectors/saltstack_cve_2021_25281/src/main/java/com/google/tsunami/plugins/detectors/saltstack/cve202125281/SaltStackCVE202125281VulnDetector.java

```diff
@@ -0,0 +1,153 @@
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.tsunami.common.net.http.HttpRequest.post;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.common.io.Resources;
+import com.google.common.net.MediaType;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.net.http.HttpStatus;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects the CVE-2021-25281 vulnerability.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "SaltStackCVE202125281VulnDetector",
+    version = "1.0",
+    description = "This detector checks for SaltStack Salt-API Unauthenticated "
+        + "Remote Command Execution vulnerability (CVE-2021-25281).",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = SaltStackCVE202125281VulnDetectorBootstrapModule.class
+)
+public class SaltStackCVE202125281VulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final String payloadFormatString;
+
+  @Inject
+  SaltStackCVE202125281VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+    String payloadFormatString = "";
+    try {
+      payloadFormatString = Resources.toString(
+          Resources.getResource(this.getClass(), "payloadFormatString.json"), UTF_8);
+    } catch (IOException e) {
+      logger.atSevere().withCause(e).log(
+          "Should never happen. Couldn't load payload resource file");
```

Comment:
Let's change this to throw an `AssertionError` instead. Logging will just swallow this error and the plugin will continue execution regardless of the error. We should probably change the other example as well.

You can just `throw new AssertionError("Couldn't load payload resource file.", e);`